core/trino-main/src/main/java/io/trino/execution/QueryStateMachine.java

```diff
@@ -983,6 +986,16 @@ public void setOutput(Optional<Output> output)
         this.output.set(output);
     }
 
+    public void setSelectColumnsLineageInfo(List<ColumnLineageInfo> selectOutputColumnsLineage)
```

Comment:
nit: Analysis can return it as `Optional<List<ColumnLineageInfo>>` right ? In this case we could directly set the lineage info.